packages/botonic-api/src/rest/routes/users.ts

```diff
@@ -1,55 +1,182 @@
+import { User } from '@botonic/core/src/models/user'
 import { Router } from 'express'
-// import { dataProviderFactory } from '../../data-provider/'
-// import { ApiGatewayManagementApi } from 'aws-sdk'
+import { checkSchema, matchedData, validationResult } from 'express-validator'
+
+import { dataProviderFactory } from '../../data-provider'
+import {
+  getOptionalSchema,
+  limitParamSchema,
+  offsetParamSchema,
+} from './validation/common'
+import {
+  userIdParamSchema,
+  userSchema,
+  userWithUserIdParamSchema,
+} from './validation/users'
 
 const router = Router()
 
 router
   .route('/')
-  .get((_, res) => {
-    res.send('GET HTTP method on user resource')
-  })
-  .post((_, res) => {
-    res.send('POST HTTP method on user resource')
+  .get(
+    checkSchema({
+      limit: limitParamSchema,
+      offset: offsetParamSchema,
+    }),
+    async (req, res) => {
+      try {
+        const errors = validationResult(req)
+        if (!errors.isEmpty()) {
+          res.status(400).send({ errors: errors.array() })
+          return
+        }
+
+        const query = matchedData(req, { locations: ['query'] })
+        const dp = dataProviderFactory(process.env.DATA_PROVIDER_URL)
+        const users = await dp.getUsers(query.limit, query.offset)
+        res.status(200).send(users)
```

Comment:
Usually when returning results in a paginated API you'd expect an object like this:

```
{
  "prev": "<link to prev page>",
  "next": "<link to next page>",
  "count": <total number of records>,
  "results": [{...user1}, {...user2}, ...]
}
```

Reply:
:+1: I'll add this into another PR with the `page`/`pageSize` refactor.